pkg/binding/definition.go

```diff
@@ -207,99 +207,115 @@ func (d *mapFromDataFieldDefinition) Apply(u *unstructured.Unstructured) (Value,
 
 type stringOfMapDefinition struct {
 	outputName string
-	path       []string
+	definition
 }
 
 var _ Definition = (*stringOfMapDefinition)(nil)
 
-func (d *stringOfMapDefinition) GetPath() []string { return d.path }
-
 func (d *stringOfMapDefinition) Apply(u *unstructured.Unstructured) (Value, error) {
-	val, ok, err := unstructured.NestedFieldNoCopy(u.Object, d.path...)
+	val, err := getValuesByJSONPath(u.Object, d.path)
 	if err != nil {
 		return nil, err
 	}
+	if len(val) != 1 {
+		return nil, fmt.Errorf("Only one value should be returned for %v but we got %v", d.path, val)
```

Comment:
Error strings should not be capitalized.
Ref. https://github.com/golang/go/wiki/CodeReviewComments#error-strings